br-webui/index.js

```diff
@@ -1840,4 +1840,40 @@ io.on('connection', function(socket) {
 		});
 
 	});
+
+	socket.on('set companion as dhcp server', function() {
+		logger.log("Companion set to dhcp server");
+
+		child_process.exec(home_dir+'/companion/scripts/config-dhcp.sh', function (error, stdout, stderr) {
+			logger.log(stdout + stderr);
+		});
+		logger.log("Done setting");
+
+	});
+	socket.on('set companion as manual', function() {
+		logger.log("Companion set to manual");
+
+		child_process.exec(home_dir+'/companion/scripts/config-manual.sh', function (error, stdout, stderr) {
+			logger.log(stdout + stderr);
+		});
+		logger.log("Done setting");
+
+	});
+	socket.on('set companion as dhcp client', function() {
```

Comment:
you should change each of these three functions to a single handler:

`socket.on('configure-network', function(configuration) {`